server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalWeightedAvg.java

```diff
@@ -88,37 +88,21 @@ public String getWriteableName() {
 
     @Override
     public InternalWeightedAvg doReduce(List<InternalAggregation> aggregations, ReduceContext reduceContext) {
-        double weight = 0;
-        double sum = 0;
-        double sumCompensation = 0;
-        double weightCompensation = 0;
+        CompensatedSum sumCompensation = CompensatedSum.newZeroInstance();
+        CompensatedSum weightCompensation = CompensatedSum.newZeroInstance();
+
         // Compute the sum of double values with Kahan summation algorithm which is more
         // accurate than naive summation.
         for (InternalAggregation aggregation : aggregations) {
             InternalWeightedAvg avg = (InternalWeightedAvg) aggregation;
-            // If the weight is Inf or NaN, just add it to the running tally to "convert" to
-            // Inf/NaN.  This keeps the behavior bwc from before kahan summing
```

Comment:
I think this inline comment is worth preserving; Let's just copy it to the same clause in the new class.